### PR TITLE
use empty list in case of null

### PR DESCRIPTION
**Pull request by honnix**
# Hey, I just made a Pull Request!

## Description

`policy.getBindings()` may return `null` if there is no binding.

## Motivation and Context

To fix NPE.

## Checklist for PR author(s)

- [x] Changes are covered by unit test
- [x] All tests pass
- [x] Code coverage check passes
- [x] Error handling is tested
- [ ] Errors are handled at the appropriate layer
- [ ] Errors that cannot be handled where they occur are propagated
- [ ] (optional) Changes are covered by system test
- [ ] Relevant documentation updated
- [x] This PR has NO breaking change to public API
- [ ] This PR has breaking change to public API and it is documented

## Checklist for PR reviewer(s)

- [x] This PR has been incorporated in release note for the coming version
- [ ] Risky changes introduced by this PR have been all considered

### CODE REVIEW

File: styx-api-service/src/main/java/com/spotify/styx/api/ServiceAccountUsageAuthorizer.java [hunk 1]
```diff
@@ -253,7 +254,7 @@ private String lookupServiceAccountProjectId(String email) {
           .orElseThrow(() -> new ResponseException(Response.forStatus(
               BAD_REQUEST.withReasonPhrase("Project does not exist: " + projectId))));
 
-      final List<String> members = policy.getBindings().stream()
+      final List<String> members = Optional.ofNullable(policy.getBindings()).orElse(Collections.emptyList()).stream()
```
Comment (NatashaL) [inline styx-api-service/src/main/java/com/spotify/styx/api/ServiceAccountUsageAuthorizer.java]: This looks duplicate. Can we move it into `memberStatus`, and have `memberStatus(principalEmail, policy)` instead?

**Comment by NatashaL** [comment]
A test would also be good to have. Otherwise it looks good

File: styx-api-service/src/main/java/com/spotify/styx/api/ServiceAccountUsageAuthorizer.java (hunk 1 above)
Comment (honnix) [reply styx-api-service/src/main/java/com/spotify/styx/api/ServiceAccountUsageAuthorizer.java]: I will add tests but would prefer doing massage before calling the method. It is the matter of contract, I think.

Comment (honnix) [reply styx-api-service/src/main/java/com/spotify/styx/api/ServiceAccountUsageAuthorizer.java]: Plus, it looks the same, but they are actually different policy/binding classes.

File: styx-api-service/src/main/java/com/spotify/styx/api/ServiceAccountUsageAuthorizer.java [hunk 2]
```diff
@@ -266,12 +267,11 @@ private String lookupServiceAccountProjectId(String email) {
           .orElseThrow(() -> new ResponseException(Response.forStatus(
               BAD_REQUEST.withReasonPhrase("Service account does not exist: " + serviceAccount))));
 
-      final List<String> members = policy.getBindings().stream()
+      final List<String> members = Optional.ofNullable(policy.getBindings()).orElse(Collections.emptyList()).stream()
           .filter(binding -> serviceAccountUserRole.equals(binding.getRole()))
           .flatMap(binding -> binding.getMembers().stream())
```
Comment (danielnorberg) [inline styx-api-service/src/main/java/com/spotify/styx/api/ServiceAccountUsageAuthorizer.java]: Might `getMembers` also return null?

File: styx-api-service/src/main/java/com/spotify/styx/api/ServiceAccountUsageAuthorizer.java (hunk 1 above)
Comment (danielnorberg) [reply styx-api-service/src/main/java/com/spotify/styx/api/ServiceAccountUsageAuthorizer.java]: Yeah I wanted to do that code deduplication but it's different types so doesn't get much nicer than this.

**Comment by danielnorberg** [comment]
👍

File: styx-api-service/src/main/java/com/spotify/styx/api/ServiceAccountUsageAuthorizer.java (hunk 2 above)
Comment (honnix) [reply styx-api-service/src/main/java/com/spotify/styx/api/ServiceAccountUsageAuthorizer.java]: :(

**Comment by Zatine** [comment]
👍

**Pull request merged**